### PR TITLE
feature : Supports presence and air quality sensor

### PR DESCRIPTION
**Pull request by khw1006**
- Supports presence sensor and air quality sensor (PM2.5)

### CODE REVIEW

File: src/fibaroAccessory.ts [hunk 1]
```diff
@@ -129,6 +129,12 @@ export class FibaroAccessory {
             this.platform.Characteristic.SecuritySystemTargetState];
           subtype = '0--';
           break;
+        case 'airQualitySensorPm25':
+          service = this.platform.Service.AirQualitySensor;
+          this.mainCharacteristics = [this.platform.Characteristic.AirQuality,
+            this.platform.Characteristic.PM2_5Density];
+          subtype = device.id + '--PM2_5';
```
Comment (mkz212) [inline src/fibaroAccessory.ts:136]: I don't think subtype is needed for the device? This is probably only for virtual things like scene, climate zone, security, etc.

File: src/fibaroAccessory.ts [hunk 2]
```diff
@@ -529,6 +541,7 @@ export class FibaroAccessory {
     service.isClimateZone = (IDs.length >= 3 && IDs[2] === 'CZ') ? true : false;
     service.isHeatingZone = (IDs.length >= 3 && IDs[2] === 'HZ') ? true : false;
     service.isOpenCloseOnly = (IDs.length >= 3 && IDs[2] === 'OPENCLOSEONLY') ? true : false;
+    service.isPM2_5Sensor = (IDs.length >= 3 && IDs[2] === 'PM2_5') ? true : false;
```
Comment (mkz212) [inline src/fibaroAccessory.ts:544]: Need to specify service.isPM2_5Sensor? Is it used anywhere after that?

File: src/getFunctions.ts [hunk 3]
```diff
@@ -507,6 +510,25 @@ export class GetFunctions {
     }
   }
 
+  getAirQuality(characteristic, _service, _IDs, properties) {
+    const v = parseFloat(properties.value);
+    // 2024 AQI for PM 2.5
+    // https://www.epa.gov/system/files/documents/2024-02/pm-naaqs-air-quality-index-fact-sheet.pdf
+    if (_service.isPM2_5Sensor) {
```
Comment (mkz212) [inline src/getFunctions.ts:517]: Need to check this? Something else also use getAirQuality function?

File: src/fibaroAccessory.ts (hunk 2 above)
Comment (mkz212) [reply src/fibaroAccessory.ts:544]: Ok it is in getAirQuality function but is it required there?

File: src/getFunctions.ts (hunk 3 above)
Comment (khw1006) [reply src/getFunctions.ts:517]: There are air quality sensors to check other things.

image

Comment (mkz212) [reply src/getFunctions.ts:517]: Ok, so this is already a preparation if they were added in the future.

File: src/fibaroAccessory.ts (hunk 2 above)
Comment (khw1006) [reply src/fibaroAccessory.ts:544]: Yes, it is used to check which type of air quality sensor it is.

File: src/getFunctions.ts (hunk 3 above)
Comment (mkz212) [reply src/getFunctions.ts:517]: > Yes, I want to support other kind of air quality sensors in future.

👍

File: src/fibaroAccessory.ts (hunk 1 above)
Comment (mkz212) [reply src/fibaroAccessory.ts:136]: Ok, now i know that this is for future, for other types of air quality sensors.

**Comment by khw1006** [comment]
@ilcato Can you review this PR?

**Review by ilcato — APPROVED** [review]

**Pull request merged**